src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1519,7 +1519,14 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
                 op2 = impSIMDPopStack(retType);
                 op1 = impSIMDPopStack(retType);
 
-                retNode = gtNewSimdCmpOpNode(GT_GE, retType, op1, op2, simdBaseJitType, simdSize,
+                genTreeOps op = GT_GE;
+                if (!varTypeIsFloating(simdBaseType))
+                {
+                    op = GT_LT;
+                    std::swap(op1, op2);
+                }
+
+                retNode = gtNewSimdCmpOpNode(op, retType, op1, op2, simdBaseJitType, simdSize,
```

Comment:
nit: Would prefer we handle this in `gtNewSimdCmpOpNode`. It ensures that any other places that need to create such nodes get the right optimization.

Reply:
Ah, makes sense